extensions/amp-ad/0.1/amp-ad-cross-domain-iframe-handler.js

```diff
@@ -107,6 +106,9 @@ export class AmpAdApiHandler {
       // If support render-start, create a race between render-start no-content
       this.adResponsePromise_ = listenForOncePromise(this.iframe_,
```

Comment:
IMO the code would be more readable if we moved to more of a message handler structure where this is an object that maps message key to handler.  Something like:

const MessageHandlers = {'render-start': this.onRenderStartMessage_, 'no-content': this.onNoContentMessage_, 'bootstrap-loaded': this.onBootStrapLoaded_};
this.adResponsePromise_ = listenForOncePromise(this.iframe_, Object.keys(MessageHandlers), this.is3p_).then(info => {
    MessageHandlers[info.data.type](info);
});